src/test/java/com/endava/internship/collections/StudentTest.java

```diff
@@ -0,0 +1,124 @@
+package com.endava.internship.collections;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.time.Month;
+
+public class StudentTest {
+    @Test
+    public void whenComparingTwoEqualStudents_EqualsShouldReturnTrue() {
+        Student ion = new Student("Ion", LocalDate.of(2002, Month.OCTOBER, 4), "student ulim");
+        Student ion2 = new Student("Ion", LocalDate.of(2002, Month.OCTOBER, 4), "student ulim");
+
+        Assertions.assertTrue(ion.equals(ion2));
+
+    }
+
+    @Test
+    public void whenComparingAStudentToItself_EqualsShouldReturnTrue() {
+        Student ion = new Student("Ion", LocalDate.of(2002, Month.OCTOBER, 4), "student ulim");
+
+        Assertions.assertTrue(ion.equals(ion));
+    }
+
+    @Test
+    public void whenComparingTwoNonEqualStudents_EqualsShouldReturnFalse() {
+        Student ion = new Student("Ion", LocalDate.of(2002, Month.OCTOBER, 4), "student ulim");
+        Student alex = new Student("Alex", LocalDate.of(2001, Month.JANUARY, 11), "student ulim");
+
+        Assertions.assertFalse(ion.equals(alex));
+    }
+
+    @Test
+    public void whenComparingAStudentToNull_EqualsShouldReturnFalse() {
+        Student ion = new Student("Ion", LocalDate.of(2002, Month.OCTOBER, 4), "student ulim");
+
+        Assertions.assertFalse(ion.equals(null));
+    }
+
+    @Test
+    public void whenComparingTwoObjectsOfDifferentClasses_EqualsShouldReturnFalse() {
+        Student ion = new Student("Ion", LocalDate.of(2002, Month.OCTOBER, 4), "student ulim");
+        String string = new String("some text");
```

Comment:
Why not use this style?
```suggestion
        String string = "some text";
```